File: C++ Projects/Quiz/questionsa.cpp
//-------------------------
//class implentation(.cpp)
//author: Noah Funderburgh
//File:questionsa.cpp
//-------------------------
#include "question.h"
#include "questionsa.h"

using namespace std;
//------------------------
questionSA::questionSA(){

}
//------------------------
questionSA::~questionSA(){

}
//------------------------
void questionSA::showQuestion(){
    cout << questionText << "?" <<  endl;
}
//------------------------
void questionSA::showAnswer(){
    cout << answerText << endl;
}
//------------------------
bool questionSA::checkAnswer(string answer){
    string answerTextLower = answerText;
    for(int i = 0;i<=answer.length();i++){
        if(answer[i] >= 65 && answer[i]<=92){
            answer[i]=answer[i]+32;
        }
    }

    for(int i = 0;i<=answerTextLower.length();i++){
        if(answerTextLower[i] >= 65 && answerTextLower[i]<=92){
            answerTextLower[i]=answerTextLower[i]+32;
        }
    }

    if(answer.at(0) == answerTextLower.at(0)){
        correct = true;
        return true;
    }
    else{
        correct = false;
        return false;
    }
}
//------------------------

File: Java Projects/Quiz/Questionsa.java
/**
* Implements classes need to add a short answer question
* Functionality to our Quiz
*
* @author  Noah Funderburgh
* @version 1.0
* @since   3/3/2020
*/

public class Questionsa extends Question{
	/**
	 * This functions will show the answer when calling the function for short answer questions.
   	 */
	void showQuestion() {
		System.out.println(questionText + "?");
	}

//-------------------------------------------------------
	/**
	 * This function checks the if userAnswer is equal to Answer by making both strings lowercase
	 * and then comparing the two to check if there equal.
	 * 
	 * @param userAnswer answer thats input by the user in the quiz
	 * @return boolean returns true if there equal else it returns false.
   	 */
	boolean checkAnswer(String userAnswer) {
		userAnswer = userAnswer.toLowerCase();
		answerText = answerText.toLowerCase();
		if(answerText.equals(userAnswer)) {
			return true;
		}
		else {
			return false;
		}
	}
}